inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/hive/PartitionLeaderElectionRunnable.java

```diff
@@ -0,0 +1,217 @@
+import org.apache.inlong.sort.standalone.utils.InlongLoggerFactory;
+import org.slf4j.Logger;
+
+/**
+ * 
+ * PartitionLeaderElectionRunnable
+ */
+public class PartitionLeaderElectionRunnable implements Runnable {
+
+    public static final Logger LOG = InlongLoggerFactory.getLogger(HiveSinkContext.class);
+    private final HiveSinkContext context;
+    private Map<String, PartitionCreateRunnable> partitionCreateMap = new ConcurrentHashMap<>();
+
+    /**
+     * Constructor
+     * 
+     * @param context
+     */
+    public PartitionLeaderElectionRunnable(HiveSinkContext context) {
+        this.context = context;
+    }
+
+    /**
+     * run
+     */
+    @Override
+    public void run() {
+        // clear stopped runnable
+        Set<String> uidPartitions = new HashSet<>();
+        for (Entry<String, PartitionCreateRunnable> entry : this.partitionCreateMap.entrySet()) {
+            if (entry.getValue().getState() != PartitionState.INIT
+                    && entry.getValue().getState() != PartitionState.CREATING) {
+                uidPartitions.add(entry.getKey());
+            }
+        }
+        uidPartitions.forEach(item -> this.partitionCreateMap.remove(item));
+        // add runnable
+        try (Connection conn = context.getHiveConnection()) {
+            Map<String, HdfsIdConfig> idConfigMap = context.getIdConfigMap();
+            ExecutorService partitionCreatePool = context.getPartitionCreatePool();
+            Statement stat = conn.createStatement();
+            for (Entry<String, HdfsIdConfig> entry : idConfigMap.entrySet()) {
+                if (hasToken(entry.getValue())) {
+                    HdfsIdConfig idConfig = entry.getValue();
+                    // get partition list of table
+                    String tableName = idConfig.getHiveTableName();
+                    ResultSet rs = stat.executeQuery("show partitions " + tableName);
+                    Set<String> partitionSet = new HashSet<>();
+                    while (rs.next()) {
+                        String strPartition = rs.getString(1);
+                        int index = strPartition.indexOf('=');
+                        if (index < 0) {
+                            continue;
+                        }
+                        partitionSet.add(strPartition.substring(index + 1));
+                    }
+                    rs.close();
+                    // close partition
+                    long currentTime = System.currentTimeMillis();
+                    long beginScanTime = currentTime
+                            - 2 * idConfig.getMaxPartitionOpenDelayHour() * HdfsIdConfig.HOUR_MS;
+                    long endScanTime = currentTime - idConfig.getPartitionIntervalMs()
+                            - context.getMaxFileOpenDelayMinute() * HiveSinkContext.MINUTE_MS;
+                    long forceCloseTime = currentTime - idConfig.getMaxPartitionOpenDelayHour() * HdfsIdConfig.HOUR_MS;
```

Comment:
The logical suggestion here is to extract a method （97-102）

Reply:
three time variables need 3 methods, the logical should not be invoked repeatly.